polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -290,21 +290,24 @@ mlir::Attribute MLIRScanner::InitializeValueByInitListExpr(mlir::Value ToInit,
         } else {
           auto PT = ToInit.getType().cast<LLVM::LLVMPointerType>();
           auto ET = PT.getElementType();
-          mlir::Type NextType;
           if (auto ST = ET.dyn_cast<LLVM::LLVMStructType>())
-            NextType = ST.getBody()[I];
+            Next = Builder.create<LLVM::GEPOp>(
+                Loc,
+                LLVM::LLVMPointerType::get(ST.getBody()[I],
+                                           PT.getAddressSpace()),
+                ToInit, llvm::ArrayRef<mlir::LLVM::GEPArg>{0, I});
           else if (auto AT = ET.dyn_cast<LLVM::LLVMArrayType>())
-            NextType = AT.getElementType();
+            Next = Builder.create<LLVM::GEPOp>(
+                Loc,
+                LLVM::LLVMPointerType::get(AT.getElementType(),
+                                           PT.getAddressSpace()),
+                ToInit, llvm::ArrayRef<mlir::LLVM::GEPArg>{0, I});
+          else if (ET.isInteger(8))
+            Next = Builder.create<LLVM::GEPOp>(
+                Loc, LLVM::LLVMPointerType::get(ET, PT.getAddressSpace()),
+                ToInit, llvm::ArrayRef<mlir::LLVM::GEPArg>{I});
```

Comment:
Would this correspond to a `void *` pointer?

Reply:
Yes. `void *` is represented as `i8*`.